extensions/wikia/Flags/controllers/FlagsController.class.php

```diff
@@ -131,26 +148,80 @@ public function getFlagsForPage() {
 			return null;
 		}
 
-		$flagModel = new Flag();
-		$flagsForPage = $flagModel->getFlagsForPage( $this->params['wikiId'], $this->params['pageId'] );
+		$flagsForPage = $this->tryGetFlagsForPageFromCache( $this->params['wikiId'], $this->params['pageId'] );
 
 		$this->setResponseData( $flagsForPage );
 	}
 
+	/**
+	 * Tries to get the data on Flags instances for the given page from cache.
+	 * If it is not cached it gets it from the database and caches it.
+	 * @param $wikiId
+	 * @param $pageId
+	 * @return bool|mixed
+	 */
+	private function tryGetFlagsForPageFromCache( $wikiId, $pageId ) {
```

Comment:
I'm not sure about naming, because finally it always returns the data, not just try ;)